rust/chains/tw_solana/src/compiler.rs

```diff
@@ -86,7 +89,8 @@ impl SolanaCompiler {
             let signature = ed25519::Signature::try_from(sign.as_slice())?;
             let pubkey = ed25519::sha512::PublicKey::try_from(pubkey.as_slice())?;
 
-            if !pubkey.verify(signature.clone(), data_to_sign.clone()) {
+            if !pubkey.verify(signature.clone(), data_to_sign.clone())
+                && signature.to_bytes() != H512::from(ZERO_SIGNATURE) {
```

Comment:
It's not optimal to parse the string as H512 on each iteration.
Could you please add `Hash<N>::is_zero`:
```rust
pub fn is_zero(&self) -> bool {
    self.0.iter().all(|byte| *byte == 0)
}
```

and use it this way:
```rust
if !pubkey.verify(signature.clone(), data_to_sign.clone()) && !signature.to_bytes().is_zero() {
    return Err();
}
```

Reply:
Sure, and done.